python/mspasspy/db/database.py

```diff
@@ -1793,9 +1793,9 @@ def read_ensemble_data(self, objectid_list, ensemble_metadata={},
                 object_type, wf_collection), 'Fatal')
 
         if object_type is TimeSeries:
-            ensemble = TimeSeriesEnsemble(len(objectid_list))
+            ensemble = TimeSeriesEnsemble(len(list(objectid_list)))
         else:
-            ensemble = SeismogramEnsemble(len(objectid_list))
+            ensemble = SeismogramEnsemble(len(list(objectid_list)))
```

Comment:
why adding the conversion to list here? @haruming  I guess it is an attempt to handle cursor, but the conversion here will consume the whole cursor, making it invalid to be referenced from in the next loop. Also, it seems this is not tested with a cursor, otherwise it should reveal the problem.